plugins/statistics.koplugin/calendarview.lua

```diff
@@ -765,19 +765,51 @@ function CalendarDayView:goToPage(page)
 end
 
 function CalendarDayView:onNextPage()
-    if not self:nextPage() and self.day_ts + 86400 < os.time() then
-        -- go to next day
-        self.day_ts = self.day_ts + 86400
-        self:setupView()
+    if not self:nextPage() and self.day_ts + 82800 < os.time() then
+        local next_day_ts = self.day_ts + 86400 + 10800 -- make sure it's the next day
+        local next_day_date = os.date("*t", next_day_ts)
+        next_day_ts = os.time({
+            year = next_day_date.year,
+            month = next_day_date.month,
+            day = next_day_date.day,
+        })
+        local current_day_date = os.date("*t", self.day_ts)
+        local current_day_ts = os.time({
```

Comment:
Is it possible that self.day_ts has not gone thru this "alignment" process ?
If yes, shouldn't it be done at the other places that won't do it, instead of redoing it each time here?